include/alpaka/warp/WarpUniformCudaHipBuiltIn.hpp

```diff
@@ -0,0 +1,152 @@
+#include <alpaka/core/BoostPredef.hpp>
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !BOOST_LANG_CUDA
+    #error If ALPAKA_ACC_GPU_CUDA_ENABLED is set, the compiler has to support CUDA!
+#endif
+
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED) && !BOOST_LANG_HIP
+    #error If ALPAKA_ACC_GPU_HIP_ENABLED is set, the compiler has to support HIP!
+#endif
+
+#include <alpaka/warp/Traits.hpp>
+
+#include <cstdint>
+
+namespace alpaka
+{
+    namespace warp
+    {
+        //#############################################################################
+        //! The GPU CUDA/HIP warp.
+        class WarpUniformCudaHipBuiltIn : public concepts::Implements<ConceptWarp, WarpUniformCudaHipBuiltIn>
+        {
+        public:
+            //-----------------------------------------------------------------------------
+            WarpUniformCudaHipBuiltIn() = default;
+            //-----------------------------------------------------------------------------
+            __device__ WarpUniformCudaHipBuiltIn(WarpUniformCudaHipBuiltIn const &) = delete;
+            //-----------------------------------------------------------------------------
+            __device__ WarpUniformCudaHipBuiltIn(WarpUniformCudaHipBuiltIn &&) = delete;
+            //-----------------------------------------------------------------------------
+            __device__ auto operator=(WarpUniformCudaHipBuiltIn const &) -> WarpUniformCudaHipBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            __device__ auto operator=(WarpUniformCudaHipBuiltIn &&) -> WarpUniformCudaHipBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            ~WarpUniformCudaHipBuiltIn() = default;
+        };
+
+        namespace traits
+        {
+            //#############################################################################
+            template<>
+            struct GetSize<
+                WarpUniformCudaHipBuiltIn>
+            {
+                //-----------------------------------------------------------------------------
+                __device__ static auto getSize(
+                    warp::WarpUniformCudaHipBuiltIn const & /*warp*/)
+                -> std::int32_t
+                {
+                    return warpSize;
+                }
+            };
+
+            //#############################################################################
+            template<>
+            struct All<
+                WarpUniformCudaHipBuiltIn>
+            {
+                //-----------------------------------------------------------------------------
+                __device__ static auto all(
+                    warp::WarpUniformCudaHipBuiltIn const & /*warp*/,
+                    std::int32_t predicate)
+                -> std::int32_t
+                {
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && BOOST_COMP_NVCC >= BOOST_VERSION_NUMBER(9, 2, 0)
```

Comment:
The test linux_clang-10_cuda-9.2_release us with this fix working, this mean that something is wrong with our CI. IMO the CI test is using the wrong CUDA version. Instead of CUDA 9.2 maybe an older installed version is used.

Reply:
https://github.com/alpaka-group/alpaka/pull/1003/checks?check_run_id=794879618#step:4:5738 is showing that we are using the second branch for CUDA < 9.2

Reply:
Turned out, I made a mistake in the check and should have checked the clang cuda version, not nvcc version. This is why we went to another branch. Investigating further now.